bioformats2raw module

CaroAMN opened the pull request:
<!--
# nf-core/modules pull request

Many thanks for contributing to nf-core/modules!

Please fill in the appropriate checklist below (delete whatever is not relevant).
These are the most common things requested on pull requests (PRs).

Remember that PRs should be made against the master branch.

Learn more about contributing: [CONTRIBUTING.md](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
-->

## PR checklist

Closes #7422 <!-- If this PR fixes an issue, please link it here! --> 

- [x] This comment contains a description of changes (with reason).
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.

**new modules: bioformats2raw**

I used now a seqera container: community.wave.seqera.io/library/bioformats2raw:0.9.4--3eec45888b3759e5
- not sure if this is correct in the module main.nf 
- nf-test passes with docker 

The tool is not in conda-forge or bioconda but from the ome channel ( open microscopy environment ) and i get errors from linting:

`The environment.yml of the module bioformats2raw is not valid: 'ome' is not one of ['bioconda', 'conda-forge'].`

camlloyd reviewed (changes requested):
Looks good! A few comments though.

In theory Wave allows any conda channel, but the linter disagrees 🧐.

camlloyd commented on modules/nf-core/bioformats2raw/main.nf:
```diff
@@ -0,0 +1,47 @@
+    def prefix = task.ext.prefix ?: "${meta.id}"
+
+    """
+    bioformats2raw \\
```
I see you used `label 'process_single'` above.

Out of interest did you try using `--max-workers $task.cpus` here?

camlloyd replied:
Looks good to me.

camlloyd commented on modules/nf-core/bioformats2raw/meta.yml:
```diff
@@ -0,0 +1,50 @@
+description: Java application to convert image file formats, including .mrxs, to an intermediate Zarr structure compatible with the OME-NGFF specification.
+keywords:
+  - zarr
+  - OME-NGFF
```
This needs to be all lower case.

camlloyd replied:
Looks good to me.

CaroAMN commented:
so i added the changes :) Thank you so much for your help @camlloyd , i added the process_high label because of the max_workers argument. I am not sure which one of the process labels would be better. 

The linter has a problem with the ome conda channel: 

> The tool is not in conda-forge or bioconda but from the ome channel ( open microscopy environment ) and i get errors from linting:
The environment.yml of the module bioformats2raw is not valid: 'ome' is not one of ['bioconda', 'conda-forge'].

But then i guess it would be ready for review again. :)

camlloyd commented on modules/nf-core/bioformats2raw/meta.yml:
```diff
@@ -0,0 +1,50 @@
+    - image:
+        type: file
+        description: bioformats supported image file format 
+        pattern: "*.{tif,tiff,svs,mrxs}"
```
Looks good to me.

camlloyd commented on modules/nf-core/bioformats2raw/meta.yml:
```diff
@@ -0,0 +1,50 @@
+    - "*ome.zarr":
+        type: directory
+        description: OME-Zarr fileset for image data storage. Specification can be found at https://ngff.openmicroscopy.org/latest/#storage-format
+        pattern: "*{ome.zarr}"
```
Looks good to me.

camlloyd commented:
> so i added the changes :) Thank you so much for your help @camlloyd , i added the process_high label because of the max_workers argument. I am not sure which one of the process labels would be better.
> 
> The linter has a problem with the ome conda channel:
> 
> > The tool is not in conda-forge or bioconda but from the ome channel ( open microscopy environment ) and i get errors from linting:
> > The environment.yml of the module bioformats2raw is not valid: 'ome' is not one of ['bioconda', 'conda-forge'].
> 
> But then i guess it would be ready for review again. :)

Added an action point regarding the emitted versions. I will open an issue regarding the linting error.

camlloyd commented:
See https://github.com/nf-core/tools/issues/3499

camlloyd commented on modules/nf-core/bioformats2raw/main.nf:
```diff
@@ -0,0 +1,47 @@
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        bioformats2raw: \$(bioformats2raw --version |& sed '1!d ; s/bioformats2raw //')
```
Looks good to me!

camlloyd commented:
> > so i added the changes :) Thank you so much for your help @camlloyd , i added the process_high label because of the max_workers argument. I am not sure which one of the process labels would be better.
> > The linter has a problem with the ome conda channel:
> > > The tool is not in conda-forge or bioconda but from the ome channel ( open microscopy environment ) and i get errors from linting:
> > > The environment.yml of the module bioformats2raw is not valid: 'ome' is not one of ['bioconda', 'conda-forge'].
> > 
> > 
> > But then i guess it would be ready for review again. :)
> 
> Added an action point regarding the emitted versions. I will open an issue regarding the linting error.

@CaroAMN you can ignore this specific linting issue for now (thanks @mirpedrol!)

camlloyd reviewed (approved):
Well done 🥇

CaroAMN commented:
> Well done 🥇

Thank you so much for your help :)!

maxulysse commented on modules/nf-core/bioformats2raw/main.nf:
```diff
@@ -0,0 +1,52 @@
+
```
```suggestion
```

maxulysse commented:
the vcf produced seems to be unstable, can you try using the nft-vcf plugin to check them out?

CaroAMN commented:
> the vcf produced seems to be unstable, can you try using the nft-vcf plugin to check them out?

mh not sure what is happening but the tests failing are for `Test Process CNVNATOR_CONVERT2VCF` which is not my module... 😱

CaroAMN commented:
> > the vcf produced seems to be unstable, can you try using the nft-vcf plugin to check them out?
> 
> mh not sure what is happening but the tests failing are for `Test Process CNVNATOR_CONVERT2VCF` which is not my module... 😱

ok never mind, issue resolved

mahesh-panchal reviewed (approved):
Looks good. It's nice we're starting to include other community channels.

edmundmiller commented:
I think there's an issue for this somewhere... https://github.com/nf-core/modules/blob/5dd32197158306d66ab90958ea1c3588bba172ec/modules/nf-core/ale/environment.yml

maxulysse commented:
Force merging it in

maxulysse commented:
> I think there's an issue for this somewhere... https://github.com/nf-core/modules/blob/5dd32197158306d66ab90958ea1c3588bba172ec/modules/nf-core/ale/environment.yml

ale is on bioconda, no?

For me this is similar to the idea with pip (cf https://github.com/nf-core/tools/issues/3461).
Aka we need to allow more stuff.
An issue has been created for that one in particular already: https://github.com/nf-core/tools/issues/3499

luiskuhn commented:
Awesome! :)

CaroAMN commented:
Thank you ! :)
